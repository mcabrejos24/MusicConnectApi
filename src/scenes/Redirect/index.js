import React from 'react';

export default function Redirect() {

    const token = window.location.hash.substr(1).split('&')[0].split("=")[1];

    if (token) {
        console.log('vijay');
        window.opener.spotifyCallback(token)
    } 

    return (
        <div className="content">
            <p>Redirect Page</p>
        </div>
    );
}
