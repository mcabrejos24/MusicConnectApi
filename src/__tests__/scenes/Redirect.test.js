import React from "react";
import { shallow } from "enzyme";
import Redirect from '../../scenes/Redirect';
import toJson from "enzyme-to-json";

describe("renders Redirect page", () => {
    let page;
    beforeEach(() => {
        page = shallow(<Redirect />);
    });
    it("Redirect loads", () => {
        page;
    });
    it("props class name matches", () => {
        expect(page.props().testData).toEqual("no-href-or-callback");
    });
    it("matches snapshot", () => {
        expect(toJson(page)).toMatchSnapshot();
    });
});

describe("tests with href, spotifiyCallback and page params", () => {
    let testHref;
    let testSpotifyCallback;
    let page;
    beforeEach(() => {
        testSpotifyCallback = (payload) => {
            return payload;
        };    
    });
    describe("tests for failed condition of no redirect", () => {
        beforeEach(() => {
            testHref = 'supposed to fail url';
            page = shallow(<Redirect testHref={testHref} testSpotifyCallback={testSpotifyCallback}/>);
        });
        it("redirect not passed into url", () => {
            expect(page.props().testData).toEqual("no-redirect-in-url");
        });
        it("matches snapshot", () => {
            expect(toJson(page)).toMatchSnapshot();
        });
    });
    
    describe("tests for failed condition of no state and code params", () => {
        beforeEach(() => {
            testHref = 'http://localhost:3000/redirect';
            page = shallow(<Redirect testHref={testHref} testSpotifyCallback={testSpotifyCallback}/>);
        });
    
        it("code and/or state not passed into url", () => {
            expect(page.props().testData).toEqual("no-code-and-or-state-in-url");
        });
        it("matches snapshot", () => {
            expect(toJson(page)).toMatchSnapshot();
        });
    });
});

describe("tests with testState param added", () => {
    let testHref;
    let testSpotifyCallback;
    let testState;
    let page;
    beforeEach(() => {
        testHref = 'http://localhost:3000/redirect?code=somecode&state=someState';
        testSpotifyCallback = (payload) => {
            return payload;
        };
    });
    describe("tests for failed condition of state not matching", () => {
        beforeEach(() => {
            testState = 'stateMeantToFail';
            page = shallow(<Redirect testHref={testHref} testSpotifyCallback={testSpotifyCallback} testState={testState}/>);
        });
    
        it("code and/or state not passed into url", () => {
            expect(page.props().testData).toEqual("state-failed-to-match");
        });
        it("matches snapshot", () => {
            expect(toJson(page)).toMatchSnapshot();
        });
    });
    
    describe("tests for correct render of Redirect page", () => {
        beforeEach(() => {
            testState = 'someState';
            page = shallow(<Redirect testHref={testHref} testSpotifyCallback={testSpotifyCallback} testState={testState}/>);
        });
        it("page got code sucessfully", () => {
            expect(page.props().testData).toEqual("render-successful");
        });
        it("has correct class", () => {
            expect(page.props().className).toEqual("content redirect-page");
        });
        it("page got code sucessfully", () => {
            expect(page.find("p").text()).toEqual("Redirect Page");
        });
        it("matches snapshot", () => {
            expect(toJson(page)).toMatchSnapshot();
        });
    });
});
