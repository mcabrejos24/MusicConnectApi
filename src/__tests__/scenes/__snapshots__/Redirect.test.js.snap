// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Redirect page matches snapshot 1`] = `
<div
  testData="no-href-or-callback"
/>
`;

exports[`tests with href, spotifiyCallback and page params tests for failed condition of no redirect matches snapshot 1`] = `
<div
  testData="no-redirect-in-url"
/>
`;

exports[`tests with href, spotifiyCallback and page params tests for failed condition of no state and code params matches snapshot 1`] = `
<div
  testData="no-code-and-or-state-in-url"
/>
`;

exports[`tests with testState param added tests for correct render of Redirect page matches snapshot 1`] = `
<div
  className="content redirect-page"
  testData="render-successful"
>
  <p>
    Redirect Page
  </p>
</div>
`;

exports[`tests with testState param added tests for failed condition of state not matching matches snapshot 1`] = `
<div
  testData="state-failed-to-match"
/>
`;
